lib/service.js

```diff
@@ -34,73 +32,51 @@ class Service extends EventEmitter {
 
   /**
    * Fetch service details from AWS
-   *
-   * @param {function} cb Callback
    */
-  _service(cb) {
-    var ecs = new AWS.ECS();
+  async _service() {
+    const ecs = new AWS.ECS();
 
-    var params = {
+    const params = {
       services: [this.options.serviceName],
-      cluster: this.options.clusterArn
+      cluster: this.options.clusterArn,
     };
 
-    ecs.describeServices(params, (err, data) => {
-      if (err) {
-        logger.error(err.message);
-        return cb(err);
-      }
-      logger.info("Retrieved service data for " + data['services'][0]['serviceArn']);
-      cb(null, data['services'][0]);
-    });
+    const data = await ecs.describeServices(params).promise();
+    logger.info(`Retrieved service data for ${data.services[0].serviceArn}`);
+    return data.services[0];
   }
 
   /**
    * Fetch health for all targets
-   *
-   * @param {function} cb Callback
    */
-  _targets(cb) {
-    var alb = new AWS.ELBv2();
-    var lbs = _.filter(this.raw.loadBalancers, (lb) => !!lb.targetGroupArn);
-
-    async.map(lbs, (lb, done) => {
-      alb.describeTargetHealth({ TargetGroupArn: lb.targetGroupArn }, (err, data) => {
-        if (err) return done(err);
-        done(null, data.TargetHealthDescriptions);
-      });
-    }, (err, results) => {
-      if (err) return cb(err);
-      cb(null, _.flatten(results));
-    });
+  async _targets() {
+    const alb = new AWS.ELBv2();
+    const lbs = this.raw.loadBalancers.filter((lb) => !!lb.targetGroupArn);
+
+    const results = await Promise.all(lbs.map(async (lb) => {
+      const data = await alb.describeTargetHealth({TargetGroupArn: lb.targetGroupArn}).promise();
+      return data.TargetHealthDescriptions;
+    }));
+
+    return _.flatten(results);
   }
 
   /**
    * Fetch Cluster Container Instances
-   *
-   * @param {function} cb Callback
    */
-  _clusterContainerInstances(cb) {
-    var ecs = new AWS.ECS();
-
-    async.waterfall([
-      (done) => {
-        var params = { cluster: this.options.clusterArn };
-        ecs.listContainerInstances(params, done);
-      },
-
-      (results, done) => {
-        var params = {
-          cluster: this.options.clusterArn,
-          containerInstances: results.containerInstanceArns
-        };
-
-        ecs.describeContainerInstances(params, (err, results) => {
-          if (err) return done(err);
-          done(null, results.containerInstances);
-        });
-      }
-    ], cb);
+  async _clusterContainerInstances() {
+    const ecs = new AWS.ECS();
+
+    let results = await ecs.listContainerInstances({
+      cluster: this.options.clusterArn,
+    }).promise();
+
+    results = await ecs.describeContainerInstances({
```

Comment:
nit: why not just make these two things separate consts with meaningful names?

Reply:
Done